--- git status ---
On branch main
Your branch is up to date with 'origin/main'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   source/humanoid_isaac_dash/humanoid_isaac_dash/data/Robots/ymboy-13dof-w-usd/.asset_hash
	modified:   source/humanoid_isaac_dash/humanoid_isaac_dash/data/Robots/ymboy-13dof-w-usd/config.yaml

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   source/humanoid_isaac_dash/humanoid_isaac_dash/assets/ymbot_boy_skate.py
	modified:   source/humanoid_isaac_dash/humanoid_isaac_dash/tasks/skating/config/ymboy_skate/skate_env_cfg_v4.py
 


--- git diff ---
diff --git a/source/humanoid_isaac_dash/humanoid_isaac_dash/assets/ymbot_boy_skate.py b/source/humanoid_isaac_dash/humanoid_isaac_dash/assets/ymbot_boy_skate.py
index 779d6e0..1d4e390 100644
--- a/source/humanoid_isaac_dash/humanoid_isaac_dash/assets/ymbot_boy_skate.py
+++ b/source/humanoid_isaac_dash/humanoid_isaac_dash/assets/ymbot_boy_skate.py
@@ -18,6 +18,7 @@ from isaaclab.assets import ArticulationCfg
 
 from humanoid_isaac_dash.assets import ISAACLAB_ASSETS_DATA_DIR
 
+from isaaclab.utils import configclass
 ##
 # Configuration
 ##
@@ -143,6 +144,7 @@ YMBOT_BOT_SKATE_14dof_CFG = ArticulationCfg(
     },
 )
 
+
 YMBOT_BOT_SKATE_13DOF_CFG = ArticulationCfg(
     spawn=sim_utils.UrdfFileCfg(
         asset_path=f"{ISAACLAB_ASSETS_DATA_DIR}/Robots/ymboy-13dof-w/ymboy_13dof_w.urdf",
@@ -242,6 +244,103 @@ YMBOT_BOT_SKATE_13DOF_CFG = ArticulationCfg(
 )
 
 
+YMBOT_BOT_SKATE_13DOF_MULTI_ROBOT_CFG = ArticulationCfg(
+    spawn=sim_utils.MultiUsdFileCfg(
+        usd_path=[
+            f"{ISAACLAB_ASSETS_DATA_DIR}/Robots/ymboy-13dof-w-usd/ymboy_13dof_w.usd",
+            f"{ISAACLAB_ASSETS_DATA_DIR}/Robots/ymboy-13dof-w-usd/ymboy_13dof_w_wheel_offset1.usd",
+            f"{ISAACLAB_ASSETS_DATA_DIR}/Robots/ymboy-13dof-w-usd/ymboy_13dof_w_wheel_offset2.usd",
+            f"{ISAACLAB_ASSETS_DATA_DIR}/Robots/ymboy-13dof-w-usd/ymboy_13dof_w_wheel_radius1.usd",
+            f"{ISAACLAB_ASSETS_DATA_DIR}/Robots/ymboy-13dof-w-usd/ymboy_13dof_w_wheel_radius2.usd",
+        ],
+        random_choice=True,
+        rigid_props=sim_utils.RigidBodyPropertiesCfg(
+            disable_gravity=None,
+            retain_accelerations=False,
+            enable_gyroscopic_forces=True,
+            linear_damping=0.0,
+            angular_damping=0.0,
+            max_linear_velocity=1000.0,
+            max_angular_velocity=1000.0,
+            max_depenetration_velocity=1.0,
+        ),
+        articulation_props=sim_utils.ArticulationRootPropertiesCfg(
+            enabled_self_collisions=False,
+            solver_position_iteration_count=8,
+            solver_velocity_iteration_count=4,
+            sleep_threshold=0.005,
+            stabilization_threshold=0.001,
+        ),
+        activate_contact_sensors=True,
+    ),
+    init_state=ArticulationCfg.InitialStateCfg(
+        pos=(0.0, 0.0, 0.7),
+        # joint_pos={".*": 0.0},
+        joint_pos={
+            ".*_hip_pitch_joint": -0.1,
+            ".*_knee_joint": 0.3,
+            ".*_ankle_pitch_joint": -0.2,
+        },
+        joint_vel={".*": 0.0},
+    ),
+    soft_joint_pos_limit_factor=0.9,
+    actuators={
+        "legs": ImplicitActuatorCfg(
+            joint_names_expr=[
+                ".*_hip_pitch_joint",
+                ".*_hip_roll_joint",
+                ".*_hip_yaw_joint",
+                ".*_knee_joint",],
+            effort_limit=94,
+            velocity_limit=15.0,
+            stiffness={
+                ".*_hip_pitch_joint": 150.0,
+                ".*_hip_roll_joint": 100.0,
+                ".*_hip_yaw_joint": 100.0,
+                ".*_knee_joint": 150.0,
+            },
+            damping={
+                ".*_hip_pitch_joint": 15, 
+                ".*_hip_roll_joint": 10, 
+                ".*_hip_yaw_joint": 10,
+                ".*_knee_joint": 15,
+            },
+            armature=0.01,
+            friction=2e-3,
+            dynamic_friction=1e-3,
+        ),
+        "feet": ImplicitActuatorCfg(
+            joint_names_expr=[".*_ankle_pitch_joint", ".*_ankle_roll_joint"],
+            effort_limit=72,
+            velocity_limit=16.0,
+            stiffness={".*_ankle_pitch_joint": 20, ".*_ankle_roll_joint": 20},
+            damping={".*_ankle_pitch_joint": 2, ".*_ankle_roll_joint": 2},
+            armature=0.01,
+            friction=2e-3,
+            dynamic_friction=1e-3,
+        ),
+        "waist": IdealPDActuatorCfg(
+            joint_names_expr=["waist.*"],
+            effort_limit_sim=94,
+            velocity_limit_sim=14,
+            stiffness=20.0,
+            damping=5.0,
+            armature=0.01,
+            # friction= 2e-3,
+            # dynamic_friction= 1e-3,
+        ),
+        "passive_pulley": ImplicitActuatorCfg(
+            joint_names_expr=[".*_pulley_.*_joint"],
+            effort_limit_sim=1000.0,
+            velocity_limit_sim=1000.0,
+            stiffness=0.0,
+            damping=0.001,
+            # friction= 2e-3,
+            # dynamic_friction = 1.5e-3,
+        ),
+    },
+)
+
 
 
 YMBOT_BOT_SKATE_12DOF_CFG = ArticulationCfg(
diff --git a/source/humanoid_isaac_dash/humanoid_isaac_dash/data/Robots/ymboy-13dof-w-usd/.asset_hash b/source/humanoid_isaac_dash/humanoid_isaac_dash/data/Robots/ymboy-13dof-w-usd/.asset_hash
index eae485e..f92d5a7 100644
--- a/source/humanoid_isaac_dash/humanoid_isaac_dash/data/Robots/ymboy-13dof-w-usd/.asset_hash
+++ b/source/humanoid_isaac_dash/humanoid_isaac_dash/data/Robots/ymboy-13dof-w-usd/.asset_hash
@@ -1 +1 @@
-136474bd3a2e66068504bcdf2fd73772
\ No newline at end of file
+224b6e8f4dd7da56c6d497995874b4a0
\ No newline at end of file
diff --git a/source/humanoid_isaac_dash/humanoid_isaac_dash/data/Robots/ymboy-13dof-w-usd/config.yaml b/source/humanoid_isaac_dash/humanoid_isaac_dash/data/Robots/ymboy-13dof-w-usd/config.yaml
index 0e7506b..5de7f7d 100644
--- a/source/humanoid_isaac_dash/humanoid_isaac_dash/data/Robots/ymboy-13dof-w-usd/config.yaml
+++ b/source/humanoid_isaac_dash/humanoid_isaac_dash/data/Robots/ymboy-13dof-w-usd/config.yaml
@@ -1,6 +1,6 @@
-asset_path: /home/ps/pan_zheng_proj/humanoid_isaac_dash_proj/source/humanoid_isaac_dash/humanoid_isaac_dash/data/Robots/ymboy-13dof-w/ymboy_13dof_w_wheel_radius2.urdf
+asset_path: /home/ps/pan_zheng_proj/humanoid_isaac_dash_proj/source/humanoid_isaac_dash/humanoid_isaac_dash/data/Robots/ymboy-13dof-w/ymboy_13dof_w_wheel_offset1.urdf
 usd_dir: /home/ps/pan_zheng_proj/humanoid_isaac_dash_proj/source/humanoid_isaac_dash/humanoid_isaac_dash/data/Robots/ymboy-13dof-w-usd
-usd_file_name: ymboy_13dof_w_wheel_radius2.usd
+usd_file_name: ymboy_13dof_w_wheel_offset1.usd
 force_usd_conversion: true
 make_instanceable: true
 fix_base: false
@@ -19,5 +19,5 @@ self_collision: false
 replace_cylinders_with_capsules: true
 collision_from_visuals: false
 ##
-# Generated by UrdfConverter on 2026-01-06 at 17:08:04.
+# Generated by UrdfConverter on 2026-01-06 at 17:12:45.
 ##
diff --git a/source/humanoid_isaac_dash/humanoid_isaac_dash/tasks/skating/config/ymboy_skate/skate_env_cfg_v4.py b/source/humanoid_isaac_dash/humanoid_isaac_dash/tasks/skating/config/ymboy_skate/skate_env_cfg_v4.py
index a45eb89..1676dad 100644
--- a/source/humanoid_isaac_dash/humanoid_isaac_dash/tasks/skating/config/ymboy_skate/skate_env_cfg_v4.py
+++ b/source/humanoid_isaac_dash/humanoid_isaac_dash/tasks/skating/config/ymboy_skate/skate_env_cfg_v4.py
@@ -19,7 +19,7 @@ from humanoid_isaac_dash.tasks.skating.velocity_env_cfg import LocomotionVelocit
 
 from isaaclab.managers import CurriculumTermCfg as CurrTerm
 # use local assets
-from humanoid_isaac_dash.assets.ymbot_boy_skate import YMBOT_BOT_SKATE_13DOF_CFG, YMBOT_BOT_SKATE_12DOF_CFG  # isort: skip
+from humanoid_isaac_dash.assets.ymbot_boy_skate import YMBOT_BOT_SKATE_13DOF_CFG, YMBOT_BOT_SKATE_13DOF_MULTI_ROBOT_CFG  # isort: skip
 
 from humanoid_isaac_dash.terrains.config.rough import ROUGH_SKATE_TERRAINS_CFG, PLAY_SKATE_TERRAINS_CFG
 
@@ -521,7 +521,7 @@ class YMBOYSkateEnvCfg(LocomotionVelocityRoughEnvCfg):
         # post init of parent
         super().__post_init__()
         # ------------------------------Sence------------------------------
-        self.scene.robot = YMBOT_BOT_SKATE_13DOF_CFG.replace(prim_path="{ENV_REGEX_NS}/Robot")
+        self.scene.robot = YMBOT_BOT_SKATE_13DOF_MULTI_ROBOT_CFG.replace(prim_path="{ENV_REGEX_NS}/Robot")
         self.scene.height_scanner.prim_path = "{ENV_REGEX_NS}/Robot/" + self.torso_link_name
         self.scene.height_scanner_base.prim_path = "{ENV_REGEX_NS}/Robot/" + self.base_link_name
 